client-src/elements/chromedash-preflight-dialog.js

```diff
@@ -0,0 +1,175 @@
+  if (!preflightDialogEl) {
+    preflightDialogEl = document.createElement('chromedash-preflight-dialog');
+    document.body.appendChild(preflightDialogEl);
+    await preflightDialogEl.updateComplete;
+  }
+  preflightDialogEl.openWithContext(
+    feature, progress, process, action, stage, feStage);
+}
+
+
+export function somePendingPrereqs(action, progress) {
+  return action.prerequisites.some(
+    itemName => !progress.hasOwnProperty(itemName));
+}
+
+
+export class ChromedashPreflightDialog extends LitElement {
+  static get properties() {
+    return {
+      feature: {type: Object},
+      progress: {type: Object},
+      process: {type: Object},
+      action: {type: Object},
+      stage: {type: Object},
+      feStage: {type: Object},
+      progress: {type: Object},
+    };
+  }
+
+  constructor() {
+    super();
+    this.feature = null;
+    this.progress = null;
+    this.process = null;
+    this.action = null;
+    this.stage = null;
+    this.feStage = null;
+    this.progress = null;
+  }
+
+  static get styles() {
+    return [
+      ...SHARED_STYLES,
+      css`
+      ol {
+        list-style: none;
+        padding: 0;
+      }
+
+      ol li {
+        margin-top: .5em;
+      }
+
+      .missing-prereqs-list {
+        padding-bottom: 1em;
+      }
+
+      .missing-prereqs-list li {
+        list-style: circle;
+        margin-left: 2em;
+      }
+
+      .edit-progress-item {
+        visibility: hidden;
+        margin-left: var(--content-padding-half);
+      }
+
+      .active .edit-progress-item,
+      .missing-prereqs .edit-progress-item,
+      .pending:hover .edit-progress-item,
+      .done:hover .edit-progress-item {
+        visibility: visible;
+      }
+    `];
+  }
+
+  openWithContext(feature, progress, process, action, stage, feStage) {
+    this.feature = feature;
+    this.progress = progress;
+    this.process = process;
+    this.action = action;
+    this.stage = stage;
+    this.feStage = feStage;
+    this.shadowRoot.querySelector('sl-dialog').show();
+  }
+
+  closeDialog() {
+    this.shadowRoot.querySelector('sl-dialog').hide();
+  }
+
+  renderEditLink(stage, feStage, pi) {
+    const featureId = this.feature.id;
+    let editEl = nothing;
+    if (pi.field) {
+      editEl = html`
+        <a class="edit-progress-item"
+           href="/guide/stage/${featureId}/${stage.outgoing_stage}/${feStage.id}#id_${pi.field}"
+           @click=${this.closeDialog}>
+          Edit
+        </a>
+      `;
+    }
+    return editEl;
+  }
+
+  makePrereqItem(itemName) {
+    for (const s of (this.process.stages || [])) {
+      for (const pi of s.progress_items) {
+        if (itemName == pi.name) {
+          return {...pi, stage: s};
+        }
+      }
+    }
+    throw new Error('prerequiste is not a defined progress item');
+  }
+
+  renderDialogContent() {
+    console.log(this);
+    if (this.feature == null) {
+      console.log('feature is null');
```

Comment:
Should this be a console.warn, and stay in the code?

Reply:
Removed